### PR TITLE
bump ubuntu and llvm

### PR DESCRIPTION
**Pull request by jorendumoulin**
Ubuntu bump to provide python 12 support for xdsl
LLVM bump to be able to use mlir functionality in tensorflow

awaiting #180 for code preparations for bump

### CODE REVIEW

File: util/container/Dockerfile [hunk 1]
```diff
@@ -2,11 +2,19 @@
...
 # SPDX-License-Identifier: Apache-2.0
 
-FROM ubuntu:22.04 AS base
+
+FROM ubuntu:24.04 AS build-bender
+
+RUN apt-get update && apt-get install -y curl build-essential
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+RUN cargo install bender --version 0.28.1
```
[the remark this answers is not in the document]
Comment (jorendumoulin) [reply util/container/Dockerfile:11]: multi-stage build magic yo

[the remark this answers is not in the document]
Comment (JosseVanDelm) [reply util/container/Dockerfile:11]: right :D

**Review by JosseVanDelm — APPROVED** [review]

**Comment by jorendumoulin** [comment]
Difference in size is 3.67 -> 3.81 GB, but this is mostly because of the ubuntu and tensorflow upgrades

**Comment by jorendumoulin** [comment]
This is ready for me. Only caveat is the `git checkout joren/bump-tensorfow` on line 96, however, without this the builds will break. So if this is allright, the plan is merge this one -> update snax-mlir -> remove the git checkout

**Review by rgantonio — APPROVED** [review]
Such work around hehe

**Pull request merged**